### PR TITLE
Replace oc_* module usage with upstream k8s module

### PR DESCRIPTION
**Pull request by mtnbikenc**
Updates ssh_bastion creation to use k8s modules.

### CODE REVIEW

File: test/aws/ssh_bastion.yml [hunk 1]
```diff
@@ -1,51 +1,213 @@
 ---
-- name: Create ssh bastion project
-  oc_project:
-    name: openshift-ssh-bastion
-    description: openshift-ssh-bastion
-    node_selector: ""
+- name: Create ssh bastion namespace
+  k8s:
```
Comment (DanyC97) [inline test/aws/ssh_bastion.yml]: w00t w00t Mr @mtnbikenc !! awesome ;)

**Comment by mtnbikenc** [comment]
352 - Cluster setup failed

/retest

File: test/aws/ssh_bastion.yml [hunk 2]
```diff
@@ -1,51 +1,205 @@
...
+        ssh_host_rsa_key: "{{ lookup('file', '../../inventory/dynamic/injected/ssh-privatekey') | b64encode }}"
+        sshd_config: "{{ lookup('file', 'files/sshd_config') | b64encode }}"
   no_log: true
 
-- name: Create ssh bastion
-  command: "oc --kubeconfig={{ kubeconfig_path }} apply -f files/"
+- name: Create ssh bastion service
+  k8s:
+    kubeconfig: "{{ kubeconfig_path }}"
+    resource_definition:
```
Comment (vrutkovs) [inline test/aws/ssh_bastion.yml]: I'd prefer `definition: "{{ lookup('template', 'files/01_service.yml') }}"` here instead to make the playbook more readable

**Comment by vrutkovs** [comment]
```
error parsing /tmp/openshift-ansible-ILVlJwo/machineset.yaml: error converting YAML to JSON: yaml: line 46: found unexpected ':'
```
probably due to 
```
- name: tag:Name
```
/retest
```

File: test/aws/scaleup.yml [hunk 3]
```diff
@@ -13,13 +13,11 @@
...
     changed_when: False
-  - name: add localhost as master
-    add_host:
-      name: localhost
-      ansible_connection: local
-      groups: masters
 
   - import_tasks: ssh_bastion.yml
+    vars:
+      ssh_bastion_namespace: "openshift-ssh-bastion"
```
Comment (vrutkovs) [inline test/aws/scaleup.yml]: These are test playbooks, not sure if this needs to be configurable

File: test/aws/ssh_bastion.yml (hunk 2 above)
Comment (mtnbikenc) [reply test/aws/ssh_bastion.yml]: Ok, I'll rework this.

File: test/aws/scaleup.yml (hunk 3 above)
Comment (mtnbikenc) [reply test/aws/scaleup.yml]: I'll remove the var.

**Comment by mtnbikenc** [comment]
374 - Cluster build failed
```
level=fatal msg="waiting for Kubernetes API: context deadline exceeded"
```
/retest

File: playbooks/init/basic_facts.yml [hunk 4]
```diff
@@ -28,27 +28,22 @@
...
     register: clustercfg
     when: kubeconfig_path is defined
     until:
-    - clustercfg.results is defined
-    - clustercfg.results.returncode is defined
-    - clustercfg.results.returncode == 0
-    - clustercfg.results.results is defined
-    - clustercfg.results.results | length > 0
+    - clustercfg.resources is defined
+    - clustercfg.resources | length > 0
```
Comment (vrutkovs) [inline playbooks/init/basic_facts.yml]: Check that `status.apiServerURL` exists - operator may not have filled this yet

Comment (mtnbikenc) [reply playbooks/init/basic_facts.yml]: Pushed a new commit to address this.

**Comment by mtnbikenc** [comment]
Rebase for merge conflicts.

**Comment by vrutkovs** [comment]
Machineset is invalid - it has `None` instead of `null` in some fields. So it would be imported, but it won't create machines

https://github.com/vrutkovs/openshift-ansible/commit/4a087871eeefbaff5d7d4386d9892a61dc865a5c mitigates this, but it seems to be a bug in machine api operator

**Comment by mtnbikenc** [comment]
Job 549:
```
Failing tests:

[Feature:Builds][pruning] prune builds based on settings in the buildconfig  should prune completed builds based on the successfulBuildsHistoryLimit setting [Suite:openshift/conformance/parallel]
[Feature:DeploymentConfig] deploymentconfigs with revision history limits [Conformance] should never persist more old deployments than acceptable after being observed by the controller [Suite:openshift/conformance/parallel/minimal]
[k8s.io] EquivalenceCache [Serial] validates pod affinity works properly when new replica pod is scheduled [Suite:openshift/conformance/serial] [Suite:k8s]
[k8s.io] EquivalenceCache [Serial] validates pod anti-affinity works properly when new replica pod is scheduled [Suite:openshift/conformance/serial] [Suite:k8s]
[k8s.io] Probing container should *not* be restarted with a /healthz http liveness probe [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD with validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-cli] Kubectl client [k8s.io] Update Demo should do a rolling update of a replication controller  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-scheduling] PodPriorityResolution [Serial] validates critical system priorities are created and resolved [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run  [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that NodeAffinity is respected if not matching [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that NodeSelector is respected if matching  [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that NodeSelector is respected if not matching  [Conformance] [Suite:openshift/conformance/serial/minimal] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that required NodeAffinity setting is respected if matching [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that taints-tolerations is respected if matching [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPredicates [Serial] validates that taints-tolerations is respected if not matching [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPreemption [Serial] validates basic preemption works [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPreemption [Serial] validates lower priority pod preemption by critical pod [Suite:openshift/conformance/serial] [Suite:k8s]
[sig-scheduling] SchedulerPreemption [Serial] validates pod anti-affinity works in preemption [Suite:openshift/conformance/serial] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
Squashed and rebased.

**Comment by jstuever** [comment]
/retest
```
Failing tests:

[Feature:Builds][Conformance] oc new-app  should succeed with a --name of 58 characters [Suite:openshift/conformance/parallel/minimal]
[sig-api-machinery] CustomResourcePublishOpenAPI [Feature:CustomResourcePublishOpenAPI] works for CRD without validation schema [Suite:openshift/conformance/parallel] [Suite:k8s]
[sig-storage] EmptyDir volumes volume on default medium should have the correct mode [NodeConformance] [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
```

**Comment by mtnbikenc** [comment]
Job 554:
Failed due to time out pulling from the registry.

**Pull request merged**